crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -771,21 +796,25 @@ fn literal() -> impl NoirParser<ExpressionKind> {
     })
 }
 
-fn fixed_array_size() -> impl NoirParser<ArraySize> {
-    filter_map(|span, token: Token| match token {
-        Token::Int(integer) => {
-            if !integer.fits_in_u128() {
-                let message = "Array sizes must fit within a u128".to_string();
-                Err(ParserError::with_reason(message, span))
-            } else {
-                Ok(ArraySize::Fixed(integer.to_u128()))
-            }
-        }
+fn try_field_to_u64(x: acvm::FieldElement, span: Span) -> Result<u64, ParserError> {
```

Comment:
If it must fit in the u128 type why not just `try_into_u128` which exists on `noir_field`?

Reply:
I decided to store array lengths as u64 instead of u128s because I can't see any array having more than 2^64 - 1 elements. Looks like the error message below needs to be updated though